src/Fixer/ClassNotation/FinalInternalClassFixer.php

```diff
@@ -177,23 +219,39 @@ private function isClassCandidate(TokensAnalyzer $tokensAnalyzer, Tokens $tokens
             return false; // ignore class; it is abstract or already final
         }
 
-        $docToken = $tokens[$tokens->getPrevNonWhitespace($index)];
+        $attributeDecision = $phpDocDecision = null;
 
-        if (!$docToken->isGivenKind(T_DOC_COMMENT)) {
-            return $this->configuration['consider_absent_docblock_as_internal_class'];
+        if ($this->checkAttributes && $attributeIndex = $tokens->getPrevTokenOfKind($index, [[CT::T_ATTRIBUTE_CLOSE]])) {
```

Comment:
Please add test case where there are 2+ classes in one file and first has attribute while second doesn't. I believe rewinding to first `CT::T_ATTRIBUTE_CLOSE` may lead to context switch, while we need to stick to class that triggered the fixer. It would be good to support such scenario, but I'm not sure if other fixers handle that properly, so if not, we can skip this.

There is already test with multiple classes, but each has attribute, so it works properly because it's optimistic scenario 😉.